### ISSUE TITLE
Exported model returns strange results on some images

### ISSUE THREAD

**Issue by xlyashuk** [PaddlePaddle/PaddleOCR#14744]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

I trained recognition V4 EN model on 1500 images of Australian car license plates (currently for Victoria state).
Training command:
```python3 ./tools/train.py -c ./en_PP-OCRv4_rec.yml  -o Global.pretrained_model=./pretrain_v4/best_accuracy.pdparams```

Recognition is always fine when I use dynamic (non exported) model like this:

```python tools/infer_rec.py -c configs/rec/PP-OCRv4/en_PP-OCRv4_rec.yml -o Global.pretrained_model=./output/rec_ppocr_v4/best_accuracy Global.infer_img=./00240.jpg```

<details>
  <summary>Here reconition is correct: 1ZL4GF</summary>
  python tools/infer_rec.py -c configs/rec/PP-OCRv4/en_PP-OCRv4_rec.yml -o Global.pretrained_model=./output/rec_ppocr_v4/latest Global.infer_img=/root/PaddleOCR/train-data/test/00489.jpg
[2025/02/20 19:43:27] ppocr WARNING: Skipping import of the encryption module.
[2025/02/20 19:43:27] ppocr INFO: Architecture :
[2025/02/20 19:43:27] ppocr INFO:     Backbone :
[2025/02/20 19:43:27] ppocr INFO:         name : PPLCNetV3
[2025/02/20 19:43:27] ppocr INFO:         scale : 0.95
[2025/02/20 19:43:27] ppocr INFO:     Head :
[2025/02/20 19:43:27] ppocr INFO:         head_list :
[2025/02/20 19:43:27] ppocr INFO:             CTCHead :
[2025/02/20 19:43:27] ppocr INFO:                 Head :
[2025/02/20 19:43:27] ppocr INFO:                     fc_decay : 1e-05
[2025/02/20 19:43:27] ppocr INFO:                 Neck :
[2025/02/20 19:43:27] ppocr INFO:                     depth : 2
[2025/02/20 19:43:27] ppocr INFO:                     dims : 120
[2025/02/20 19:43:27] ppocr INFO:                     hidden_dims : 120
[2025/02/20 19:43:27] ppocr INFO:                     kernel_size : [1, 3]
[2025/02/20 19:43:27] ppocr INFO:                     name : svtr
[2025/02/20 19:43:27] ppocr INFO:                     use_guide : True
[2025/02/20 19:43:27] ppocr INFO:             NRTRHead :
[2025/02/20 19:43:27] ppocr INFO:                 max_text_length : 25
[2025/02/20 19:43:27] ppocr INFO:                 nrtr_dim : 384
[2025/02/20 19:43:27] ppocr INFO:         name : MultiHead
[2025/02/20 19:43:27] ppocr INFO:     Transform : None
[2025/02/20 19:43:27] ppocr INFO:     algorithm : SVTR_LCNet
[2025/02/20 19:43:27] ppocr INFO:     model_type : rec
[2025/02/20 19:43:27] ppocr INFO: Eval :
[2025/02/20 19:43:27] ppocr INFO:     dataset :
[2025/02/20 19:43:27] ppocr INFO:         data_dir : ./train_data
[2025/02/20 19:43:27] ppocr INFO:         label_file_list : ['./train_data/val_list.txt']
[2025/02/20 19:43:27] ppocr INFO:         name : SimpleDataSet
[2025/02/20 19:43:27] ppocr INFO:         transforms :
[2025/02/20 19:43:27] ppocr INFO:             DecodeImage :
[2025/02/20 19:43:27] ppocr INFO:                 channel_first : False
[2025/02/20 19:43:27] ppocr INFO:                 img_mode : BGR
[2025/02/20 19:43:27] ppocr INFO:             MultiLabelEncode :
[2025/02/20 19:43:27] ppocr INFO:                 gtc_encode : NRTRLabelEncode
[2025/02/20 19:43:27] ppocr INFO:             RecResizeImg :
[2025/02/20 19:43:27] ppocr INFO:                 image_shape : [3, 48, 320]
[2025/02/20 19:43:27] ppocr INFO:             KeepKeys :
[2025/02/20 19:43:27] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_gtc', 'length', 'valid_ratio']
[2025/02/20 19:43:27] ppocr INFO:     loader :
[2025/02/20 19:43:27] ppocr INFO:         batch_size_per_card : 128
[2025/02/20 19:43:27] ppocr INFO:         drop_last : False
[2025/02/20 19:43:27] ppocr INFO:         num_workers : 4
[2025/02/20 19:43:27] ppocr INFO:         shuffle : False
[2025/02/20 19:43:27] ppocr INFO: Global :
[2025/02/20 19:43:27] ppocr INFO:     cal_metric_during_train : True
[2025/02/20 19:43:27] ppocr INFO:     character_dict_path : ppocr/utils/en_dict.txt
[2025/02/20 19:43:27] ppocr INFO:     checkpoints : None
[2025/02/20 19:43:27] ppocr INFO:     debug : False
[2025/02/20 19:43:27] ppocr INFO:     distributed : False
[2025/02/20 19:43:27] ppocr INFO:     epoch_num : 50
[2025/02/20 19:43:27] ppocr INFO:     eval_batch_step : [0, 2000]
[2025/02/20 19:43:27] ppocr INFO:     infer_img : /root/PaddleOCR/train-data/test/00489.jpg
[2025/02/20 19:43:27] ppocr INFO:     infer_mode : False
[2025/02/20 19:43:27] ppocr INFO:     log_smooth_window : 20
[2025/02/20 19:43:27] ppocr INFO:     max_text_length : 25
[2025/02/20 19:43:27] ppocr INFO:     pretrained_model : ./output/rec_ppocr_v4/latest
[2025/02/20 19:43:27] ppocr INFO:     print_batch_step : 10
[2025/02/20 19:43:27] ppocr INFO:     save_epoch_step : 10
[2025/02/20 19:43:27] ppocr INFO:     save_inference_dir : None
[2025/02/20 19:43:27] ppocr INFO:     save_model_dir : ./output/rec_ppocr_v4
[2025/02/20 19:43:27] ppocr INFO:     save_res_path : ./output/rec/predicts_ppocrv3.txt
[2025/02/20 19:43:27] ppocr INFO:     use_gpu : True
[2025/02/20 19:43:27] ppocr INFO:     use_space_char : True
[2025/02/20 19:43:27] ppocr INFO:     use_visualdl : False
[2025/02/20 19:43:27] ppocr INFO: Loss :
[2025/02/20 19:43:27] ppocr INFO:     loss_config_list :
[2025/02/20 19:43:27] ppocr INFO:         CTCLoss : None
[2025/02/20 19:43:27] ppocr INFO:         NRTRLoss : None
[2025/02/20 19:43:27] ppocr INFO:     name : MultiLoss
[2025/02/20 19:43:27] ppocr INFO: Metric :
[2025/02/20 19:43:27] ppocr INFO:     ignore_space : False
[2025/02/20 19:43:27] ppocr INFO:     main_indicator : acc
[2025/02/20 19:43:27] ppocr INFO:     name : RecMetric
[2025/02/20 19:43:27] ppocr INFO: Optimizer :
[2025/02/20 19:43:27] ppocr INFO:     beta1 : 0.9
[2025/02/20 19:43:27] ppocr INFO:     beta2 : 0.999
[2025/02/20 19:43:27] ppocr INFO:     lr :
[2025/02/20 19:43:27] ppocr INFO:         learning_rate : 0.0005
[2025/02/20 19:43:27] ppocr INFO:         name : Cosine
[2025/02/20 19:43:27] ppocr INFO:         warmup_epoch : 5
[2025/02/20 19:43:27] ppocr INFO:     name : Adam
[2025/02/20 19:43:27] ppocr INFO:     regularizer :
[2025/02/20 19:43:27] ppocr INFO:         factor : 3e-05
[2025/02/20 19:43:27] ppocr INFO:         name : L2
[2025/02/20 19:43:27] ppocr INFO: PostProcess :
[2025/02/20 19:43:27] ppocr INFO:     name : CTCLabelDecode
[2025/02/20 19:43:27] ppocr INFO: Train :
[2025/02/20 19:43:27] ppocr INFO:     dataset :
[2025/02/20 19:43:27] ppocr INFO:         data_dir : ./train_data/
[2025/02/20 19:43:27] ppocr INFO:         ds_width : False
[2025/02/20 19:43:27] ppocr INFO:         ext_op_transform_idx : 1
[2025/02/20 19:43:27] ppocr INFO:         label_file_list : ['./train_data/train_list.txt']
[2025/02/20 19:43:27] ppocr INFO:         name : MultiScaleDataSet
[2025/02/20 19:43:27] ppocr INFO:         transforms :
[2025/02/20 19:43:27] ppocr INFO:             DecodeImage :
[2025/02/20 19:43:27] ppocr INFO:                 channel_first : False
[2025/02/20 19:43:27] ppocr INFO:                 img_mode : BGR
[2025/02/20 19:43:27] ppocr INFO:             RecConAug :
[2025/02/20 19:43:27] ppocr INFO:                 ext_data_num : 2
[2025/02/20 19:43:27] ppocr INFO:                 image_shape : [48, 320, 3]
[2025/02/20 19:43:27] ppocr INFO:                 max_text_length : 25
[2025/02/20 19:43:27] ppocr INFO:                 prob : 0.5
[2025/02/20 19:43:27] ppocr INFO:             RecAug : None
[2025/02/20 19:43:27] ppocr INFO:             MultiLabelEncode :
[2025/02/20 19:43:27] ppocr INFO:                 gtc_encode : NRTRLabelEncode
[2025/02/20 19:43:27] ppocr INFO:             KeepKeys :
[2025/02/20 19:43:27] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_gtc', 'length', 'valid_ratio']
[2025/02/20 19:43:27] ppocr INFO:     loader :
[2025/02/20 19:43:27] ppocr INFO:         batch_size_per_card : 96
[2025/02/20 19:43:27] ppocr INFO:         drop_last : True
[2025/02/20 19:43:27] ppocr INFO:         num_workers : 8
[2025/02/20 19:43:27] ppocr INFO:         shuffle : True
[2025/02/20 19:43:27] ppocr INFO:     sampler :
[2025/02/20 19:43:27] ppocr INFO:         divided_factor : [8, 16]
[2025/02/20 19:43:27] ppocr INFO:         first_bs : 96
[2025/02/20 19:43:27] ppocr INFO:         fix_bs : False
[2025/02/20 19:43:27] ppocr INFO:         is_training : True
[2025/02/20 19:43:27] ppocr INFO:         name : MultiScaleSampler
[2025/02/20 19:43:27] ppocr INFO:         scales : [[320, 32], [320, 48], [320, 64]]
[2025/02/20 19:43:27] ppocr INFO: profiler_options : None
[2025/02/20 19:43:27] ppocr INFO: train with paddle 3.0.0-beta1 and device Place(gpu:0)
W0220 19:43:27.676575 34990 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 8.9, Driver API Version: 12.6, Runtime API Version: 11.8
W0220 19:43:27.677101 34990 gpu_resources.cc:164] device: 0, cuDNN Version: 8.7.
[2025/02/20 19:43:29] ppocr INFO: load pretrain successful from ./output/rec_ppocr_v4/latest
[2025/02/20 19:43:29] ppocr INFO: infer_img: /root/PaddleOCR/train-data/test/00240.jpg
[2025/02/20 19:43:29] ppocr INFO:        result: 1ZL4GF 0.9992775917053223
[2025/02/20 19:43:29] ppocr INFO: success!
</details>

Then I export model:

```python3 tools/export_model.py -c configs/rec/PP-OCRv4/en_PP-OCRv4_rec.yml -o Global.pretrained_model=output/rec_ppocr_v4/best_accuracy  Global.save_inference_dir=./inference/en_PP-OCRv4_rec/```

And strange thing happen: for 5% of recognitions OCR returns real rubbish even on very nice images.

I tried with V4 and V3 models and results are the same: dynamic works and exported fails on the same images.

### 🏃‍♂️ Environment (运行环境)

```
bash
OS: Ubuntu 22.04
Environment: Bash
Python: 3.11.9
PaddleOCR: latest
PaddlePaddle: 3.0-beta
Install: git
RAM: 64GB
CPU: NVIDIA 4090
CPU: AMD Threadripper 5990x
CUDA: 11.8
```

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

To test exported model I created file my_infer.py inside PaddleOCR dir:

```
from paddleocr import PaddleOCR

ocr = PaddleOCR(rec_model_dir='./inference/en_PP-OCRv4_rec/', use_angle_cls=True, lang='en')
result = ocr.ocr('./00240.jpg', cls=True, det=False)
print(result)
```

Run: ```python my_infer.py```

![Works with this image](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_20_50/00231.jpg)
Result is ***1YF7LB*** both for exported and dynamic models.
![Fails with this image](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_21_22/00240.jpg)
Result is ***IB1ZL*** for exported model and ***1ZL4GF*** for dynamic model.

I tested on 700 images which were not the part of training dataset. And got around 30-35 images with such strange errors. All of these images return correct result for dynamic model.

Here are the links to reproduce the issue:
[Training dataset](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/19/17_02_37/au_vic_dataset.zip)
[Image where exported model works](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_20_50/00231.jpg)
[Image where exported model fails](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_21_22/00240.jpg)
[Link to trained dynamic model](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_19_31/best_model_dynamic.zip)
[Link to exported model](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_19_54/en_PP-OCRv4_rec.zip)
[Link to YML config used for training](https://s3.eu-central-003.backblazeb2.com/ai-uploads/uploads/2025/02/21/19_20_18/en_PP-OCRv4_rec.yml)

I downloaded pretrained model from here:
https://paddleocr.bj.bcebos.com/PP-OCRv4/english/en_PP-OCRv4_rec_train.tar

Dictionary is the same for training and inference. Shapes are the same as well.
I tried different batch sizes (16, 32, 64, 128) but the problem remains.

I even planned to use dynamic model instead of exported model (despite it's size), but I don't see any example how to do that in code.

What could be wrong?

**Comment by xlyashuk** [comment]
I'm quite surprised that this issue is marked as completed. The problem described in this issue really happens and I see no way to solve it.

**Comment by xlyashuk** [comment]
@mattheliu Thanks for the suggestion. I can call those scripts and they work well. However I need to integrate the OCR into my Python code and I don't see any straightforward way to do that. Scripts are complex and have lots of parameters. Do you know how to use dynamic model from the Python code? 

@amitlisha Thanks, unfortunately I cannot reopen the issue as it is closed by the admin. 
I have just tried this code: ocr.ocr(image_path, det=False, rec=True), but the issue is still there. Dynamic model works well and exported fails strangely on 5% of the dataset.

**Comment by xlyashuk** [comment]
I have retested this case with latest PaddleOCR repository and PaddlePaddle 3.0 and CUDA 12.6.

`python -m pip install paddlepaddle-gpu==3.0.0 -i https://www.paddlepaddle.org.cn/packages/stable/cu126/`

But results are the same, on some good images exported model still returns random text.

**Comment by Dat0901** [comment]
I encounter this as well, when use best_accuracy, it work well, but after export model, sometimes the model return none, sometimes feel like random letters

**Comment by zhangyubo0722** [comment]
Is there a log about parameter mismatch when exporting?

**Comment by Dat0901** [comment]
> Is there a log about parameter mismatch when exporting?

I have fix this issue of when export model the accuracy drop. 
There are two things I notice,

First check if the trained weight version and the config file are the same version. If not it will have some warning when export.
Second when using exported model, need to add preprocess arguments like the config file.

Just like @zhangyubo0722 said, there will be parameter mismatch when using different version of weight and config file.

**Comment by xlyashuk** [comment]
This is what I get when I export the model:

Run export:
`
python3 tools/export_model.py -c ./en_PP-OCRv4_rec.yml -o Global.pretrained_model=output/rec_ppocr_v4/latest  
Global.save_inference_dir=./inference/en_PP-OCRv4_rec/
`

Output:
`/venv/main/lib/python3.12/site-packages/paddle/utils/cpp_extension/extension_utils.py:711: UserWarning: No ccache found. Please be aware that recompiling all source files may be required. You can download and install ccache from: https://github.com/ccache/ccache/blob/master/doc/INSTALL.md
  warnings.warn(warning_message)`
`[2025/05/21 16:50:21] ppocr WARNING: Skipping import of the encryption module.`
`W0521 16:50:22.960832  7463 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 8.9, Driver API Version: 12.8, Runtime API Version: 12.6
W0521 16:50:22.961292  7463 gpu_resources.cc:164] device: 0, cuDNN Version: 9.5.`
`[2025/05/21 16:50:24] ppocr INFO: load pretrain successful from output/rec_ppocr_v4/latest`
`[2025/05/21 16:50:24] ppocr INFO: Export inference config file to ./inference/en_PP-OCRv4_rec/inference.yml
Skipping import of the encryption module`
`[2025/05/21 16:50:25] ppocr INFO: inference model is saved to ./inference/en_PP-OCRv4_rec/inference`

In my opinion output looks good.

When export parameters are different then all results are screwed, and in my case (as I mentioned above) only ~5% has some random data.
